### PR TITLE
Update german translation

### PR DESCRIPTION
**Pull request by Aaron-Ritter**
This pull request takes care of:
* Full translation of German including: all already translated and missing translation properties
* All German official Locale
* Removing superseded de translation

Closes https://github.com/FusionAuth/fusionauth-issues/issues/2199

### How to handle messages_de.properties
As mentioned we remove the messages_de.properties for now. Or we could update the general readme with: `If you want to have German defined in your FusionAuth Template please choose one of the German Locale properties files.`

### How to handle Language and locale specific comments with update.rb
We had some specific translation guidelines / rules added as a comment in the German Locale files, when running the update.rb script it overwrites theses with the messages.properties original. How should we handle these language specific notes?

### How to handle German (Italy) - DONE
The current Locale configuration mentions Geman (Italy), I think this is because there is a small region in Italy https://en.wikipedia.org/wiki/South_Tyrol bordering Austria which talks German, wikipedia and a few other websites e.g. https://www.suedtirol.de/reisefuehrer/s/sprachen-in-suedtirol.html references German as an official Language.

We can just create a copy of de_AT to de_IT for now.

### Pending Changes
There is an additional issue pending on the completion of this pull request, we will continue with this work once this request is completed.

https://github.com/FusionAuth/fusionauth-issues/issues/2251

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -82,7 +89,12 @@ theme
 missing-translations
 ├── message_ar.properties    Arabic
 ├── message_da.properties    Danish
-├── message_de.properties    German
+├── message_de_AT.properties    German (Austria)
```
Comment (mooreds) [inline README.md:92]: These will only be present if there are any missing translations (you can see that `cz` is all done, so is not present). Please remove the German entries.

Other than that, looks good to go.

Comment (mooreds) [reply README.md:92]: These lines under `missing-translations`...

Comment (Aaron-Ritter) [reply README.md]: included in the de_IT commit.

**Comment by mooreds** [comment]
> We had some specific translation guidelines / rules added as a comment in the German Locale files, when running the update.rb script it overwrites theses with the messages.properties original. How should we handle these language specific notes?

Hmmm. Options I can see:

* change update.rb to ignore comments
* move the comments to the README in a separate section
* ??

What would you recommend, @Aaron-Ritter ?

**Comment by Aaron-Ritter** [comment]
@mooreds I've added a way on how to document this in the README. If it gets to big it could be separated in to different files e.g. README_de.md or something along the line.

**Pull request merged**